src/sentry/utils/strings.py

```diff
@@ -27,23 +28,24 @@
 """
 )
 
-
-def unicode_escape_recovery_handler(err):
-    try:
-        value = err.object[err.start : err.end].decode("utf-8")
-    except UnicodeError:
-        value = ""
-    return value, err.end
-
-
-codecs.register_error("unicode-escape-recovery", unicode_escape_recovery_handler)
+INVALID_ESCAPE = re.compile(
+    r"""
+(?<!\\)              # no backslash behind
+((?:\\\\)*\\)        # odd number of backslashes
+(?!x[0-9a-fA-F]{2})  # char escape: \x__
+(?!u[0-9a-fA-F]{4})  # char escape: \u____
+(?!U[0-9a-fA-F]{8})  # char escape: \U________
+(?![0-7]{1,3})       # octal escape: \_, \__, \___
+(?![\\'"abfnrtv])    # other escapes: https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals
+""",
+    re.VERBOSE,
+)
 
 
-def unescape_string(value):
+def unescape_string(value: str):
```

Comment:
Is there any way to type hint this to be specifically a raw string? re.sub will try to process backslash escapes in its 2nd posarg if it's a regular string and fail with stuff like `\x`.

Reply:
in python object-space there's no such thing as a "raw string" -- strings are just strings (the difference in "raw"-ness is only how it appears in the source and is parsed by the ast engine)

```pycon
>>> r"wat\xwat"
'wat\\xwat'
>>> r"wat\xwat" == "wat\\xwat"
True
```